docs/DeveloperGuide.md

```diff
@@ -1,18 +1,133 @@
+* Receiving all command line inputs from the user
+* Displaying all system outputs to the user
+
+The UI consists of the following classes: 
+* `UI` - Reads user input and displays system output
+* `MessageBank` - Stores all standard system output messages
+
+### Storage component
+
+The `Storage` component is responsible for:
+* Managing the respective data files
+* Storing all system logs generated from the application
+
+The Storage consists of the following classes: 
+* `Storage` - Stores all user profile information in respective data files
+* `AppLogger` - Stores all system logs generated from the application
+
+
+### Parser component
+
+The `Parser` component is responsible for:
+* Reading user input and breaking it down to machine-readable code
+
+The Parser consists of the following classes: 
+* `Parser` - Breaks down inputs into machine-readable code
+
+### Profile component
+
+The `Profile` component is responsible for:
+* Storing all user profile information
+
+The Profile consists of the following classes: 
+* `Profile` - 
+* `DailyFoodRecord` - 
+
+### Food component
+
+The `Food` component is responsible for:
+* 
+
+The Food consists of the following classes: 
+* `Food` - 
+* `FoodNutritionInfo` - 
+
+
+## Implementation
+
+{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### 1. [Proposed] Record Meal Feature
+#### 1.1 Proposed implementation
+The record feature is facilitated by `RecordMealCommand`. It extends `Command` and overrides `execute()` and `saveResults()`
+
+Given below is an example usage scenario and how the record mechanism behaves at each step.
+
+Step 1.  The `ui` object gets user input and sends it to the `parser`. The `parser` then parses the original input into a standard `command`
+and returns it. The command type is decided by the `commandPrompt`. The `RecordMealCommand` has `record-meal` as its commandPrompt. 
+During the process a new`RecordMealCommand` object will be generated and returned to the `Logic` component.
+
+### Step1. Generate command
+![Step1. Diagram](/docs/images/Record_step1.png)
+
+Step 2.  The `Logic` composition calls the `execute()` method of the `RecordMealCommand` object. 
+
+During the process of execution, the command object will generate a `ArrayList<Food> foodList`, which maintains all `Food` items listed in the user input. 
+If a certain kind of `Food` can be found in the database of `FoodNuritionInfo`, then this food item will be fetched from the database. 
+Else a new food object will be generated simply with `foodName` without all the nutrition info.
+
+During execution, the command object will try to get a `record` object of the class `DailyFoodRecord`. The date of record specified by the user. 
+If no record of that day is found in profile, it will automatically generate a new record of that day. Then with `foodList` and `record`, can call 
+the method `record.recordMeals()`.
```

Comment:
The use of `this with that can call func` might sound a little too informal and the calling of functions could be standardized by `class#method()`. Perhaps something like this will be better? 
> If no record of that day is found in profile, it will automatically generate a new record of that day. Together with `foodList` and `record`, a call will be made to `record#recordMeals()`.

Here's a sample from Week 10:
![image](https://user-images.githubusercontent.com/32288711/77891812-1faf3300-72a4-11ea-9e20-7dd2e8ba313a.png)